applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C

```diff
@@ -0,0 +1,174 @@
+  =========                 |
+  \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox
+   \\    /   O peration     | Website:  https://openfoam.org
+    \\  /    A nd           | Copyright (C) 2024 OpenFOAM Foundation
+     \\/     M anipulation  |
+-------------------------------------------------------------------------------
+License
+    This file is part of OpenFOAM.
+
+    OpenFOAM is free software: you can redistribute it and/or modify it
+    under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    OpenFOAM is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+    for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with OpenFOAM.  If not, see <http://www.gnu.org/licenses/>.
+
+\*---------------------------------------------------------------------------*/
+
+#include "ExaCA.H"
+#include "Time.H"
+#include "fvMesh.H"
+#include "addToRunTimeSelectionTable.H"
+#include "volFields.H"
+#include "volPointInterpolation.H"
+#include "interpolation.H"
+
+#include "polyTopoChangeMap.H"
+#include "polyMeshMap.H"
+#include "polyDistributionMap.H"
+
+// * * * * * * * * * * * * * * Static Data Members * * * * * * * * * * * * * //
+
+namespace Foam
+{
+namespace functionObjects
+{
+    defineTypeNameAndDebug(ExaCA, 0);
+
+    addToRunTimeSelectionTable
+    (
+        functionObject,
+        ExaCA,
+        dictionary
+    );
+}
+}
+
+
+// * * * * * * * * * * * * * Private Member Functions  * * * * * * * * * * * //
+
+void Foam::functionObjects::ExaCA::correct()
+{
+    searchEngine_.correct();
+}
+
+// * * * * * * * * * * * * * * * * Constructors  * * * * * * * * * * * * * * //
+
+Foam::functionObjects::ExaCA::ExaCA
+(
+    const word& name,
+    const Time& runTime,
+    const dictionary& dict
+)
+:
+    fvMeshFunctionObject(name, runTime, dict),
+    dict_(dict),
+    outputPath_
+    (
+        mesh_.time().globalPath()
+       /writeFile::outputPrefix
+       /(mesh_.name() != polyMesh::defaultRegion ? mesh_.name() : word())
+       /name
+    ),
+    searchEngine_(mesh_),
+    vpi_(volPointInterpolation::New(mesh_)),
+    T_(mesh_.lookupObject<VolField<scalar>>("T"))
+{
+    read(dict);
+}
+
+
+// * * * * * * * * * * * * * * * * Destructor  * * * * * * * * * * * * * * * //
+
+Foam::functionObjects::ExaCA::~ExaCA()
+{}
+
+
+// * * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * //
+
+bool Foam::functionObjects::ExaCA::read(const dictionary& dict)
+{
+    return true;
+}
+
+
+Foam::wordList Foam::functionObjects::ExaCA::fields() const
+{
+    return wordList::null();
+}
+
+
+bool Foam::functionObjects::ExaCA::execute()
+{
+    if (mesh_.time().writeTime())
```

Comment:
are you sure you want to tie the `ExaCA` write to the main write? the function objects class has its own execute/write controls and some users might want `ExaCA` data more often than the other fields

Reply:
Yeah, this was just a verification exercise that the volPointInterpolation weights were bring properly updated with AMR. The ExaCA FunctionObject isn't doing anything useful at the moment, other than getting the code structure in place

Reply:
ok cool, thanks for the clarification